### ISSUE TITLE
[VL] window function error

### ISSUE THREAD

**Issue by FelixYBW** [apache/incubator-gluten#6600]
### Backend

VL (Velox)

### Bug description

```
24/07/26 00:58:09 ERROR [Driver] datasources.FileFormatWriter: Aborting job c89ecb16-71d7-49b6-8963-cd58dd9873e1.
org.apache.spark.sql.catalyst.analysis.UnresolvedException: Invalid call to dataType on unresolved object
	at org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute.dataType(unresolved.scala:160)
	at org.apache.spark.sql.types.StructType$.$anonfun$fromAttributes$1(StructType.scala:548)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
```

the write columns are all string and bigint. except one date column which is used as partition column.

@JkSelf is the date columun the reason?

### Spark version

Spark-3.2.x

**Comment by LoseYSelf** [comment]
what is the sql and the table metadata?

**Comment by FelixYBW** [comment]
The error is caused by window function:

```
sum(a) OVER (
        PARTITION BY b,
        c
        ORDER BY date RANGE BETWEEN 6 preceding AND CURRENT ROW
      ) 
```

**Comment by zml1206** [comment]
Can you assign this issue to me, I want to try to solve it, thank you. @FelixYBW

**Comment by JkSelf** [comment]
I have disabled the date type in https://github.com/apache/incubator-gluten/pull/6637. @zml1206, if you have the time, you are welcome to continue supporting the date type in window range frame. Thank you for your contributions.

**Comment by FelixYBW** [comment]
@JkSelf Is there dependency of your current streaming window PR?

**Comment by FelixYBW** [comment]
@JkSelf Looks the issue isn't fixed by 6600

**Comment by zml1206** [comment]
https://github.com/apache/incubator-gluten/blob/51d0a37faafc0457d594d180f7787e8c0da1f518/backends-velox/src/test/scala/org/apache/gluten/execution/TestOperator.scala#L377

The UT verification here has passed. Are there any special circumstances that have been missed? Can you provide the test sql? @FelixYBW

**Comment by FelixYBW** [comment]
@JkSelf Did you verify the SQL I shared? failed from my test.

**Comment by JkSelf** [comment]
@FelixYBW I tested the fix in my local environment and confirmed that the issue has been resolved. However, I encountered the following exception during execution `select sum(impression_count_last7days ) from (     SELECT sum(impression_count) OVER (         PARTITION BY user_id,         surface         ORDER BY date RANGE BETWEEN 6 preceding AND CURRENT ROW       ) AS impression_count_last7days     from d233 )"`


```
24/07/26 00:58:09 ERROR [Driver] datasources.FileFormatWriter: Aborting job c89ecb16-71d7-49b6-8963-cd58dd9873e1.
org.apache.spark.sql.catalyst.analysis.UnresolvedException: Invalid call to dataType on unresolved object
	at org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute.dataType(unresolved.scala:160)
	at org.apache.spark.sql.types.StructType$.$anonfun$fromAttributes$1(StructType.scala:548)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
```

@zml1206 Do you have time to look at this issue? If not,  I can follow up. Thanks.

**Comment by zml1206** [comment]
> Do you have time to look at this issue? If not, I can follow up. Thanks.

I'll try to reproduce and solve it. @JkSelf

**Comment by FelixYBW** [comment]
fixed by #6803
